specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2024-05-15-preview/networkSecurityPerimeter.json

```diff
@@ -0,0 +1,455 @@
+                "modelAsString": true
+              }
+            }
+          }
+        },
+        "profile": {
+          "description": "Network Security Perimeter profile",
+          "readOnly": true,
+          "type": "object",
+          "properties": {
+            "name": {
+              "description": "Name of the resource",
+              "type": "string"
+            },
+            "accessRulesVersion": {
+              "description": "Current access rules version",
+              "type": "number"
+            },
+            "accessRules": {
+              "description": "List of Access Rules",
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/NspAccessRule"
+              },
+              "x-ms-identifiers": [
+                "name"
+              ]
+            },
+            "diagnosticSettingsVersion": {
+              "description": "Diagnostic settings version",
+              "type": "number"
+            },
+            "enabledLogCategories": {
+              "description": "Enabled logging categories",
+              "type": "array",
+              "items": {
+                "description": "Logging category",
+                "type": "string"
+              }
+            }
+          }
+        }
+      }
+    },
+    "ProvisioningIssue": {
+      "description": "Describes provisioning issue for given NetworkSecurityPerimeterConfiguration",
+      "type": "object",
+      "readOnly": true,
+      "properties": {
+        "name": {
+          "description": "Name of the issue",
+          "type": "string"
+        },
+        "properties": {
+          "description": "Properties of provisioning issue",
+          "type": "object",
+          "readOnly": true,
+          "properties": {
+            "issueType": {
+              "description": "Type of issue",
+              "enum": [
+                "Unknown",
+                "ConfigurationPropagationFailure"
+              ],
+              "type": "string",
+              "x-ms-enum": {
+                "name": "IssueType",
+                "modelAsString": true
+              }
+            },
+            "severity": {
+              "description": "Severity of the issue.",
+              "enum": [
+                "Warning",
+                "Error"
+              ],
+              "type": "string",
+              "x-ms-enum": {
+                "name": "Severity",
+                "modelAsString": true
+              }
+            },
+            "description": {
+              "description": "Description of the issue",
+              "type": "string"
+            }
+          }
+        }
+      }
+    },
+    "NspAccessRule": {
+      "description": "Information of Access Rule in Network Security Perimeter profile",
+      "type": "object",
+      "readOnly": true,
+      "properties": {
+        "name": {
+          "description": "Name of the resource",
+          "type": "string"
+        },
```

Comment:
any format restrictions or regex patterns, min max length that can be followed ?

Reply:
No regex patterns have been specified per the NSP design: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-networking/sdn-dbansal/azure-virtual-network-manager/azure-virtual-network-manager/develop/design-docs/nsp/network-security-perimeter-design-specification